src/execution_plan/execution_plan.c

```diff
@@ -742,10 +733,10 @@ ExecutionPlan *ExecutionPlan_UnionPlans(ResultSet *result_set, AST *ast) {
 	plan->segment_count = union_count;
 	plan->query_graph = NULL;
 	plan->record_map = raxNew();
```

Comment:
`plan->is_union = true;`

Reply:
i wonder why it was removed after rebase, and moreover, how the tests passed without it. I leave it for now, but it might be redundant.

Reply:
I think `is_union` is valuable, we can use it in the optimiser phase to quickly understand if we need to apply optimisation on the different segments.